### DISCUSSION TITLE
Recommended way to use Vitest for testing components / composables?

### DISCUSSION THREAD

**Discussion by mbrodt** [nuxt/nuxt#18821] [Questions]
Hi,

I've just installed Vitest in my Nuxt project to get started with some tests, but I can't really determine the optimal way to go about testing my components or composables. 

Because of auto import, simply importing a composable in a name.test.js file doesn't work - and even if I explicitly import the composable, it won't be able to resolve the autoimports that composable is making.

It also seems like there's a few different approaches being worked on, like [@nuxt/test-utils](https://nuxt.com/docs/getting-started/testing) and [nuxt-vitest](https://github.com/danielroe/nuxt-vitest). So I'm just curious what the recommended path is, and if there's any good ways to solve the autoimport issues?

Appreciate any responses 🙌

**Comment by sduduzog** [comment]
`@nuxt/test-utils` is good for testing your whole app i.e. hitting endpoints renders correct text etc.

Using `nuxt-vitest` and/or just `vitest` with `vue-test-utils` is good for testing components in isolation. I assume this is what you're most likely looking for

**Reply by mbrodt** [reply]
Thanks for the answer! The biggest problem I've run into is still the auto importing. I would expect to be able to import a composable in my test and then call it - but maybe that's just not possible because the test isn't running in a `<script setup>` environment?

**Reply by sduduzog** [reply]
Not extensively but for a bit, I managed to just configure any auto imported non-component items as globals via vitest setup. But I believe nuxt-vitest was created to address just this. I haven't gotten a chance yet to play with it

See this issue for context https://github.com/nuxt/nuxt/issues/13077

and of course https://twitter.com/danielcroe/status/1590490011501142017
